universal-application-tool-0.0.1/app/views/LoginForm.java

```diff
@@ -12,13 +12,18 @@
   public Content render(Http.Request request) {
     return htmlContent(
         body(
+            h1("Error: You are not logged in")
+                .withCondHidden(!request.queryString("message").orElse("").equals("login")),
             h1("Log In"),
             form(
                     makeCsrfTokenInputTag(request),
                     textField("uname", "username", "Username"),
                     passwordField("pwd", "password", "Password"),
                     submitButton("login", "Submit"))
                 .withMethod("POST")
-                .withAction("/callback?client_name=FormClient")));
+                .withAction("/callback?client_name=FormClient"),
```

Comment:
suggestion (blocking): please use the reverse router `org.pac4j.play.routes.CallbackController.callback().url()`